tests/schema/test_storage.py

```diff
@@ -0,0 +1,245 @@
+    out = tester.read(right, tmp_path, lazy, validation="allow")
+
+    # Assert
+    spy.assert_called_once()
+    assert_frame_equal(df.lazy(), out.lazy())
+
+
+@pytest.mark.parametrize("tester", TESTERS)
+@pytest.mark.parametrize("lazy", [True, False])
+def test_read_write_parquet_validation_forbid_no_schema(
+    tester: SchemaStorageTester, tmp_path: Path, lazy: Literal[True | False]
+) -> None:
+    # Arrange
+    schema = create_schema("test", {"a": dy.Int64()})
+    df = schema.create_empty()
+    tester.write_untyped(df, tmp_path, lazy)
+
+    # Act
+    with pytest.raises(
+        ValidationRequiredError,
+        match=r"without validation: no schema to check validity",
+    ):
+        tester.read(schema, tmp_path, lazy, validation="forbid")
+
+
+@pytest.mark.parametrize("tester", TESTERS)
+@pytest.mark.parametrize("lazy", [True, False])
+def test_read_write_parquet_validation_forbid_invalid_schema(
+    tester: SchemaStorageTester, tmp_path: Path, lazy: Literal[True | False]
+) -> None:
+    # Arrange
+    right = create_schema("test", {"a": dy.Int64(), "b": dy.String()})
+    wrong = create_schema("wrong", {"x": dy.Int64(), "y": dy.String()})
+    df = right.create_empty()
+    tester.write_typed(wrong, df, tmp_path, lazy=lazy)
+
+    # Act / Assert
+    with pytest.raises(
+        ValidationRequiredError,
+        match=r"without validation: current schema does not match",
+    ):
+        tester.read(right, tmp_path, lazy, validation="forbid")
+
+
+# --------------------------------- VALIDATION "SKIP" -------------------------------- #
+
+
+@pytest.mark.parametrize("tester", TESTERS)
+@pytest.mark.parametrize("lazy", [True, False])
+def test_read_write_parquet_validation_skip_no_schema(
+    tester: SchemaStorageTester,
+    tmp_path: Path,
+    mocker: pytest_mock.MockerFixture,
+    lazy: Literal[True | False],
+) -> None:
+    # Arrange
+    schema = create_schema("test", {"a": dy.Int64()})
+    df = schema.create_empty()
+    tester.write_untyped(df, tmp_path, lazy)
+
+    # Act
+    spy = mocker.spy(schema, "validate")
+    tester.read(schema, tmp_path, lazy, validation="skip")
+
+    # Assert
+    spy.assert_not_called()
+
+
+@pytest.mark.parametrize("tester", TESTERS)
+@pytest.mark.parametrize("lazy", [True, False])
+def test_read_write_parquet_validation_skip_invalid_schema(
+    tester: SchemaStorageTester,
+    tmp_path: Path,
+    mocker: pytest_mock.MockerFixture,
+    lazy: Literal[True | False],
+) -> None:
+    # Arrange
+    right = create_schema("test", {"a": dy.Int64(), "b": dy.String()})
+    wrong = create_schema("wrong", {"x": dy.Int64(), "y": dy.String()})
+    df = right.create_empty()
+    tester.write_typed(wrong, df, tmp_path, lazy=lazy)
+    # Act
+    spy = mocker.spy(right, "validate")
+    tester.read(right, tmp_path, lazy, validation="skip")
+
+    # Assert
+    spy.assert_not_called()
+
+
+# ---------------------------- DELTA LAKE SPECIFICS ---------------------------------- #
```

Comment:
[nit] This applies to everywhere: could you ensure empty newlines after these comments? Otherwise, they are interpreted to belong to the first method below :pp